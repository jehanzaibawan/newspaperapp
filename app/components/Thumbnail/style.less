.npApp_thumbnail {
  display: flex;
  background-color: #9a9a9a;
  border-radius: 50%;
  width: 3.5rem;
  height: 3.5rem;
  align-self: center;
}