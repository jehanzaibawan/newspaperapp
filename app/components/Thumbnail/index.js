/**
 *
 * Thumbnail
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import './style.less';

const Thumbnail = props => (
  <div className="npApp_thumbnail">{props.source}</div>
);

Thumbnail.propTypes = {
  source: PropTypes.string,
};

Thumbnail.defaultProps = {
  source: <span>&nbsp;</span>, // TODO: Replace space with the image
};

export default memo(Thumbnail);
