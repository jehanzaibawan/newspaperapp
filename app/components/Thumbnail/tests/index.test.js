/**
 *
 * Tests for Thumbnail
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Thumbnail from '../index';

describe('<Thumbnail />', () => {
  it('Should render and match the snapshot', () => {
    const props = {
      source: 'fake-image-source',
    };

    const {
      container: { firstChild },
    } = render(<Thumbnail {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
});
