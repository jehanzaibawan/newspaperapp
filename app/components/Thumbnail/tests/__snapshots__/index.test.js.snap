// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thumbnail /> Should render and match the snapshot 1`] = `
<div
  class="npApp_thumbnail"
>
  fake-image-source
</div>
`;
